ini lookup - add case sensitive option

samdoran opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
By setting `ConfigParser.optionxform` to `to_native`, all keys are passed through that function and case is preserved. The [default behavior](https://docs.python.org/3.9/library/configparser.html#configparser.ConfigParser.optionxform) returns a lower case version of the key name rather than preserving the case.

Fixes #74601.
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
`lib/ansible/plugins/lookup/ini.py`
<!--- Write the short name of the module, plugin, task or feature below -->

##### ADDITIONAL INFORMATION

Waiting on #74629 before adding tests to avoid merge conflicts and duplicating the test changes already done there.

On Python 2, duplicate key names are silently ignored. If two key names in a section differ only in case, in Python 2 the last one defined in the file would set the value for the key.

samdoran committed:
ini lookup - add case sensitive option

bcoca commented on lib/ansible/plugins/lookup/ini.py:
```diff
@@ -37,6 +37,12 @@
+        description:
+          Whether key names read from C(file) should be case sensitive. On Python 3,
+          this prevents duplicate key errors that are silently ignored in Python 2.
+        default: False
```
no py2 on controller, so no need to specify

samdoran replied:
I updated the description.

samdoran committed:
Remove Python 2 specific wording in description

samdoran committed:
Add tests for case-sensitive option

samdoran committed:
Minor test corrections

The key name is first, not the section name.

samdoran committed:
Add case-sensitive properties file tests

This doesn't increase coverage but seems like a thing we should have a test for

samdoran committed:
Run all playbooks from a single file
